_episodes/08-hygiene.md

```diff
@@ -84,6 +84,14 @@ the tool shapes the hand that shapes the tool.
 > *   87 Turing Avenue
 > *   January 25, 1971
 > *   the XY coordinate (0.5, 3.3)
+>
+> > ## Solution
+> > New Zealand is the only atomic value.
+> > All the others contain more than one piece of information which should be stored separately:
+> > - House number, street name
+> > - Month, day, year
```

Comment:
Most databases have a `DATE` datatype so I'm inclined to say that it would be "atomic". Date columns appear in the `Visited` table above.

Reply:
Date is one of those issues that have caused people to write a lot of papers from `Edgar Codd`, definition of Atomicity.
like this article says http://www.databasedesign-resource.com/atomic-database-values.html
Do we want to break down the phone number to CountryCode-AreaCode- Prefix-lineNumber. To m,e it will depend on data am dealing with. I think the `New Zealand` is a pure atomic value.

Reply:
I feel this a good question and we could add something for the users to reflect about.

Reply:
However above clearly goes with a `DATE` field:
> we could [...] rewrite the `Visited` table like this: `id`, `lat`, `long`, `dated`
I'm not sure there's a point arguing for anything different from the SQL data types.

Reply:
@gcapes do you want to go ahead and add date to the answer and we get this in

Reply:
How about adding it with a brief caveat/discussion point? How about something like this:
> There is a `DATE` datatype, so this would ordinarily be considered atomic. However, there might be situations where we want to keep the month, day, and year as separate variables

Reply:
@gcapes, I think that will be a great explanation.

Reply:
I don't think there is any reason to ever keep those as separate. SQL has functions to extract one of those fields from the `DATE` field, so I really don't see a point.

Reply:
Final answer?
I can certainly see the logic for keeping dates as a `DATE` type, but the question asks whether the values are atomic. Atomic is subjective, and I think this merits some discussion in the solution.
>The first rule is that every value should be atomic, i.e., not contain parts that we might want to work with separately.

We might want to work with e.g. year or month separately, so for the lesson to be internally consistent, we can't just assert that dates are atomic. Best practice is something else, and it might be so easy to extract the date or month that you'd never want to store them separately (I don't know, I'm totally new to SQL).